core/src/main/java/nu/yona/server/subscriptions/entities/BuddyConnectMessage.java

```diff
@@ -4,49 +4,122 @@
  *******************************************************************************/
 package nu.yona.server.subscriptions.entities;
 
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
 import java.util.UUID;
+import java.util.stream.Collectors;
 
+import javax.persistence.Column;
 import javax.persistence.Entity;
 import javax.persistence.Transient;
 
 import nu.yona.server.crypto.seckey.SecretKeyUtil;
+import nu.yona.server.device.entities.DeviceBase;
+import nu.yona.server.device.entities.UserDevice;
 import nu.yona.server.messaging.entities.BuddyMessage;
 
 @Entity
 public abstract class BuddyConnectMessage extends BuddyMessage
 {
+	private static final int UUID_LENGTH = 36;
+	private static final int BOOLEAN_LENGTH = 5;
+	private static final int MAX_NUM_OF_DEVICES = 10;
+	private static final double ENCRYPTION_OVERHEAD_FACTOR = 1.2; // This is a guess
```

Comment:
This factor should move in a different class.

Reply:
Currently, it's an estimate only. If we move it elsewhere, it might be used more widely and might prove to be wrong. I favor keeping at the place we use it.